CMakeLists.txt

```diff
@@ -48,6 +48,8 @@ if(MSVC)
     # remove cmake's default /W3 to avoid warning D9025
     string(REGEX REPLACE "/W3" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /permissive- /W4 /WX")
+    # Add full path to __FILE__ macro (for unit-tests)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /FC")
```

Comment:
Please provide the unit-tests with the needed filepath via a macro in cmake instead of passing /FC:
```cmake
set(MY_TESTING_DIR testing)
cmake_path(ABSOLUTE_PATH MY_TESTING_DIR NORMALIZE OUTPUT_VARIABLE MY_TESTING_DIR)
target_compile_definitions(whatever_target PRIVATE "MY_TESTING_DIR=${MY_TESTING_DIR}")
```

I'd rather diagnostic messages be more brief and not specific to the build environment-- consider a message from `src\utils\path.c` vs `D:\a\openomf\openomf\src\utils\path.c`.

Reply:
Sure, I don't mind. I don't use MSVC anyway :D